src/NuGet.Core/NuGet.PackageManagement/Projects/PackagesConfigNuGetProject.cs

```diff
@@ -60,11 +58,9 @@ public PackagesConfigNuGetProject(string folderPath, Dictionary<string, object>
                 throw new ArgumentNullException(nameof(folderPath));
             }
 
-            TargetFramework = GetMetadata<NuGetFramework>(NuGetProjectMetadataKeys.TargetFramework);
-
             PackagesConfigPath = Path.Combine(folderPath, "packages.config");
 
-            var projectName = GetMetadata<string>(NuGetProjectMetadataKeys.Name);
+            var projectName = Metadata[NuGetProjectMetadataKeys.Name];
```

Comment:
Ideally project name should be distinct parameter of the constructor. But if it's too hard to change its signature consider getting the value from `metadata` parameter or from `InternalMetadata`. In the worst case add a comment.